java/core/src/main/java/com/microsoft/bond/BondedBondType.java

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package com.microsoft.bond;
+
+import java.io.IOException;
+import java.util.HashMap;
+
+/**
+ * Implements the {@link BondType} contract for bonded container data type.
+ * @param <TStruct> the class of the underlying struct value
+ */
+public final class BondedBondType<TStruct extends BondSerializable> extends BondType<Bonded<TStruct>> {
+
+    /**
+     * The name of the type as it appears in Bond schemas.
+     */
+    public static final String TYPE_NAME = "bonded";
+
+    private final StructBondType<TStruct> valueType;
+
+    // restrict instantiation to the current package
+    BondedBondType(StructBondType<TStruct> valueType) {
+        super(-valueType.hashCode());
+        this.valueType = valueType;
+    }
+
+    /**
+     * Retrieves the underlying value type descriptor.
+     *
+     * @return the underlying value type descriptor
+     */
+    public final BondType<TStruct> getValueType() {
+        return this.valueType;
+    }
+
+    @Override
+    public final String getName() {
+        return TYPE_NAME;
+    }
+
+    @Override
+    public final String getQualifiedName() {
+        return TYPE_NAME;
+    }
+
+    @Override
+    public final BondDataType getBondDataType() {
+        return this.valueType.getBondDataType();
+    }
+
+    @Override
+    public final Class<Bonded<TStruct>> getValueClass() {
+        // can't do direct cast
+        @SuppressWarnings("unchecked")
+        Class<Bonded<TStruct>> valueClass = (Class<Bonded<TStruct>>) (Class<?>) Bonded.class;
+        return valueClass;
+    }
+
+    @Override
+    public final Class<Bonded<TStruct>> getPrimitiveValueClass() {
+        return null;
+    }
+
+    @Override
+    public final boolean isNullableType() {
+        return false;
+    }
+
+    @Override
+    public final boolean isGenericType() {
+        return true;
```

Comment:
IsGenericType means what literally "generic" means, and bonded<> is a generic type (although native to bond) as well as list<>, map<,> and Box<>. Please refer to the API documentation.

---
In reply to: [123110230](https://github.com/Microsoft/bond/pull/502#discussion_r123110230) [](ancestors = 123110230)